sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -351,7 +348,15 @@ def create_entity(
                 **kwargs
             )
             return _trim_service_metadata(metadata)
-        except ResourceNotFoundError as error:
+        except HttpResponseError as error:
+            try:
+                if error.model.additional_properties["odata.error"]["code"] == "PropertiesNeedValue":
+                    if entity.get("PartitionKey") is None:
+                        raise ValueError("PartitionKey must be present in an entity")
+                    if entity.get("RowKey") is None:
+                        raise ValueError("RowKey must be present in an entity")
+            except AttributeError:
+                raise error
```

Comment:
Let's wait till my batch PR is merged - then you'll be able to replace this code with:
```
from ._error import _decode_error, _reraise_error
...
except HttpResponseError as error:
    decoded = _decode_error(error)
    if decoded.error_code == "PropertiesNeedValue":
        ....
    _reraise_error(decoded)
```

Reply:
You didn't want to update to use the decoded error responses?